core/server/master/src/main/java/alluxio/master/meta/ServerConfigurationChecker.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.meta;
+
+import alluxio.wire.ConfigProperty;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * This class is responsible for checking server-side configuration.
+ */
+public class ServerConfigurationChecker {
+  /** Map from a node id to its configuration. */
+  private Map<Long, List<ConfigProperty>> mConfMap;
+  /** Set that contains the ids of lost nodes. */
+  private Set<Long> mLostNodes;
+
+  /**
+   * Constructs a new {@link ServerConfigurationChecker}.
+   */
+  public ServerConfigurationChecker() {
+    init();
+  }
+
+  /**
+   * Initializes the default {@link ServerConfigurationChecker}.
+   */
+  public synchronized void init() {
+    mConfMap = new HashMap<>();
```

Comment:
Minor: We can make `mConfMap` and `mLostNodes` final if we instead use `mConfMap.clear()` and `mLostNodes.clear()`. This also makes it easier for the garbage collector to collect the old configurations.